src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php

```diff
@@ -0,0 +1,197 @@
+        );
+
+        return $editableCategory;
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     *
+     * @return array
+     */
+    private function getCoverImage(CategoryId $categoryId)
+    {
+        $imageType = 'jpg';
+        $image = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '.' . $imageType;
+
+        $imageTag = ImageManager::thumbnail(
+            $image,
+            'category' . '_' . $categoryId->getValue() . '.' . $imageType,
+            350,
+            $imageType,
+            true,
+            true
+        );
+
+        $imageSize = file_exists($image) ? filesize($image) / 1000 : '';
+
+        if (empty($imageTag) || empty($imageSize)) {
+            return null;
+        }
+
+        return [
+            'size' => sprintf('%skb', $imageSize),
+            'path' => $this->imageTagSourceParser->parse($imageTag),
+        ];
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     *
+     * @return array
+     */
+    private function getThumbnailImage(CategoryId $categoryId)
+    {
+        $image = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '.jpg';
+        $imageTypes = ImageType::getImagesTypes('categories');
+
+        $thumb = '';
+        $imageTag = '';
+        $formatted_small = ImageType::getFormattedName('small');
+        foreach ($imageTypes as $k => $image_type) {
+            if ($formatted_small == $image_type['name']) {
+                $thumb = _PS_CAT_IMG_DIR_ . $categoryId->getValue() . '-' . $image_type['name'] . '.jpg';
+                if (is_file($thumb)) {
+                    $imageTag = ImageManager::thumbnail(
+                        $thumb,
+                        'category_' . (int) $categoryId->getValue() . '-thumb.jpg',
+                        (int) $image_type['width'],
+                        'jpg',
+                        true,
+                        true
+                    );
+                }
+            }
+        }
+
+        if (!is_file($thumb)) {
+            $thumb = $image;
+            $imageTag = ImageManager::thumbnail($image, 'category_' . $categoryId->getValue() . '-thumb.jpg', 125, 'jpg', true, true);
+            ImageManager::resize(_PS_TMP_IMG_DIR_ . 'category_' . $categoryId->getValue() . '-thumb.jpg', _PS_TMP_IMG_DIR_ . 'category_' . $categoryId->getValue() . '-thumb.jpg', (int) $image_type['width'], (int) $image_type['height']);
+        }
+
+        $thumbSize = file_exists($thumb) ? filesize($thumb) / 1000 : false;
+
+        if (empty($imageTag) || false === $thumbSize) {
+            return null;
+        }
+
+        return [
+            'size' => sprintf('%skb', $thumbSize),
+            'path' => $this->imageTagSourceParser->parse($imageTag),
+        ];
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     *
+     * @return array
+     */
+    private function getMenuThumbnailImages(CategoryId $categoryId)
+    {
+        $menuThumbnails = [];
+
+        for ($i = 0; $i < 3; ++$i) {
```

Comment:
Why 3?

Reply:
its maximum number of menu thumbnails category can have, i cant think of a good place to put this constant? somewhere under `Domain/Category` :thinking: ? @matks wdyt?

Reply:
:+1: for const even if I don't understand why there is a limit :P